tests/unit/test_utils.py

```diff
@@ -356,6 +380,64 @@ def test_unpack_zip(self, data):
         unzip_file(test_file, self.tempdir)
         self.confirm_files()
 
+    def test_unpack_zip_failure(self):
+        """
+        Test unpacking a *.zip with file containing .. path
+        and expect exception
+        """
+        test_zip = self.make_zip_file('test_zip.zip',
+                                      ['regular_file.txt',
+                                       os.path.join('..', 'outside_file.txt')])
+        with pytest.raises(
+                InstallationError,
+                match=r'.*trying to install outside target directory.*'):
+            unzip_file(test_zip, self.tempdir)
+
+    def test_unpack_zip_success(self):
+        """
+        Test unpacking a *.zip with regular files,
+        no file will be installed outside target directory after unpack
+        so no exception raised
+        """
+        test_zip = self.make_zip_file(
+            'test_zip.zip',
+            ['regular_file1.txt',
+             os.path.join('dir', 'dir_file1.txt'),
+             os.path.join('dir', '..', 'dir_file2.txt')])
+        try:
+            unzip_file(test_zip, self.tempdir)
+        except InstallationError:
+            pytest.fail("Unexpected InstallationError")
+
+    def test_unpack_tar_failure(self):
+        """
+        Test unpacking a *.tar with file containing .. path
+        and expect exception
+        """
+        test_tar = self.make_tar_file('test_tar.tar',
+                                      ['regular_file.txt',
+                                       os.path.join('..', 'outside_file.txt')])
+        with pytest.raises(
+                InstallationError,
+                match=r'.*trying to install outside target directory.*'):
+            untar_file(test_tar, self.tempdir)
+
+    def test_unpack_tar_success(self):
+        """
+        Test unpacking a *.tar with regular files,
+        no file will be installed outside target directory after unpack
+        so no exception raised
+        """
+        test_tar = self.make_tar_file(
+            'test_tar.tar',
+            ['regular_file1.txt',
+             os.path.join('dir', 'dir_file1.txt'),
+             os.path.join('dir', '..', 'dir_file2.txt')])
+        try:
+            untar_file(test_tar, self.tempdir)
+        except InstallationError:
+            pytest.fail("Unexpected InstallationError")
```

Comment:
You don’t need to handle an exception to fail the test. Just let it raise on its own, and same comment below.

Reply:
@cjerdonek thanks, this is now fixed